commands/kubernetes.go

```diff
@@ -338,6 +366,67 @@ func tryUpdateKubeconfig(kube do.KubernetesService, clusterID, clusterName strin
 	}
 }
 
+// RunKubernetesClusterUpgrade upgrades an existing cluster to a new version.
+func RunKubernetesClusterUpgrade(c *CmdConfig) error {
+	if len(c.Args) == 0 {
+		return doctl.NewMissingArgsErr(c.NS)
+	}
+	clusterID, err := clusterIDize(c.Kubernetes(), c.Args[0])
+	if err != nil {
+		return err
+	}
+
+	version, available, err := getUpgradeVersionOrLatest(c, clusterID)
+	if err != nil {
+		return err
+	}
+	if !available {
+		notice("cluster is already up-to-date - no upgrades available")
+		return nil
+	}
+
+	kube := c.Kubernetes()
+	err = kube.Upgrade(clusterID, version)
+	if err != nil {
+		return err
+	}
+
+	notice("upgrading cluster to version %v", version)
+	return nil
+}
+
+func getUpgradeVersionOrLatest(c *CmdConfig, clusterID string) (string, bool, error) {
+	version, err := c.Doit.GetString(c.NS, doctl.ArgClusterVersionSlug)
+	if err != nil {
+		return "", false, err
+	}
+	if version != "" && version != defaultKubernetesLatestVersion {
+		return version, true, nil
+	}
+	versions, err := c.Kubernetes().GetUpgrades(clusterID)
+	if err != nil {
+		return "", false, fmt.Errorf("no version flag provided and unable to lookup the latest version from the API: %v", err)
+	}
+	switch len(versions) {
+	case 0:
+		return "", false, nil
+	case 1:
+		return versions[0].Slug, true, nil
+	}
+
+	releases, err := latestReleases(versions)
+	if err != nil {
+		return "", false, err
+	}
+	i, err := versionMaxBy(releases, func(v do.KubernetesVersion) string {
```

Comment:
Not relevant at the moment, but in the case where both patch and minor version upgrade are available this would pick the minor upgrade rather than the patch upgrade right? This seems in line with what `latest` represents, but I feel like it'd be better/safer for the customer to default to the latest patch version.

Reply:
Good call - we should try to choose the latest available upgrade within the same minor version if possible. Updated the code to do that, and also fixed a subtle bug in the existing latest version selection code.

Reply:
I would rather stick with the industry standard meaning of `latest`, which is, well, `latest`, not restricted to minor or patch (unless there's some special understanding in the k8s community). I think it would be great to also offer a `latest-minor` and a `latest-patch` as well, but let's not change the commonly understood meaning of `latest`.